src/storage/gcloud-storage/gcp-storage.service.ts

```diff
@@ -119,6 +121,45 @@ export class GCPStorageService {
         return ok(true);
     }
 
+    private async compressFile(file: File): Promise<Buffer> {
+        const chunks: Buffer[] = [];
+
+        for await (const chunk of file.stream()) {
+            chunks.push(chunk as Buffer);
```

Comment:
Ah, are you having to cast this because typescript thinks its not a buffer? If thats the case, maybe we go with `Buffer.isBuffer(chunk) ? chunk : Buffer.from(chunk)` ?